### ISSUE TITLE
Reasonable Training Loss

### ISSUE THREAD

**Issue by joyyang1215** [tinghuiz/SfMLearner#27]
Hi, I have three questions shown as follows:
1. How low the value of the loss can result in reasonable depth prediction?
I've tried to train on other dataset (ViDRILO), so I'm not quite sure if 100K iterations of training is enough.

2. Still wonder about where the log files are saved. Does the log file look like **events.out.tfevents.1505446260.joy**
`tensorboard --logdir=/path/to/tensorflow/log/files --port=8888`

when launching the https://localhost:8888, errors show up.
`E0916 00:12:21.511262 Thread-1 _internal.py:87] 127.0.0.1 - - [16/Sep/2017 00:12:21] code 400, message Bad request syntax ("\x16\x03\x01\x00\xc0\x01\x00\x00\xbc\x03\x03\x87\xf8Z'#5\x05[\x0f/\x89\xaf\x80G\x07\x98\xad.\xcb\xeah\xdeh\x1br\x87>\xc5\x90+\xed\xd3\x00\x00\x1c\x8a\x8a\xc0+\xc0/\xc0,\xc00\xcc\xa9\xcc\xa8\xc0\x13\xc0\x14\x00\x9c\x00\x9d\x00/\x005\x00")`

3. Where is the pose model saved? 
After training, I just got one model (depth model).

**Comment by yangengt123** [comment]
Hi joyyang,
I am also exploring the code. 

For your second question, I believe what the authors indicate is a file named "evets.out.tfevents.XXXX" which should appear in your checkpoints directory during the training process.

However, did you try to run the test command ? I got two problems here.
1. For the depth testing, when I run 
$ python test_kitti_depth.py --dataset_dir /path/to/raw/kitti/dataset/ --output_dir /path/to/output/directory --ckpt_file /path/to/pre-trained/model/file/
A error saying  "Failed to find any matching files for /home/fuy34/SfMLearner-master/models/" appears, no matter using my own trained checkpoints or the sample they provided. Which directory shall we input in ckpt_file? I assume it should be the folder saving the checkpoints, but it seems it does not work.

2. For the pose testing,  what is the number of test sequence means? Shall we download an additional test sequences for that?   I cannot find any folder contains a file like "**/sequences/09/times.txt" in the raw KITTI dataset.

Thanks

**Comment by joyyang1215** [comment]
Hi yangengt,
I've only tested depth and pose prediction with pre-trained models provided by the author. And it works.

For your first question, 
Have you assigned target model (ex: model-190532) for the ckpt_file parameter?
`--ckpt_file models/model-190532`

For second question,
I guess that the **test_sequence** is just one of the sequences in KITTI.
As KITTI takes too long to download, so I have no idea how the whole dataset looks like.
I think there should be a python file to convert **2011_09_26/2011_09_26_drive_0001_sync/image_02/timestamps.txt** to **/sequences/09/times.txt**, but could not find it.
So, I just rewrite the **test_kitti_depth.py**,

**Comment by yangengt123** [comment]
Thank you!
This works for my first question. 

While for the second one, I think the problem is the difference between the dataset I used and the one they used. There is a code in load_image function that author ask a path named" image_02" particularly, but the sequences I downloaded only have "image_01" and "image_00" and mine are in gray. By modifying their code, it works. 

Good luck for your project!

**Comment by black107** [comment]
Hi yangengt,
For your second question,
In the pose testing, the author used the dataset of KITTI Odometry,
http://www.cvlibs.net/datasets/kitti/eval_odometry.php 
you can download the odometry development kit to read the included ReadMe txt for more information.
you will find it also originate from the kitti raw data.

**Comment by yangengt123** [comment]
Hi black107,
Thanks. The problem is that it seems the authors used the color version, while I downloaded the gray version. But it fine. I have tested on it, and it turned out to work well.      \
Thank you!
